config/permissions.php

```diff
@@ -417,7 +417,32 @@
         ),
     ),
 
-
+    'Custom Fieldsets' => array(
```

Comment:
Is this really necessary, do you think? I would imagine that if you are permitted to edit custom fields, you should be able to decide what fieldsets they belong in as well. (Not to mention that they won't have any effect unless you can also edit models).

Reply:
```
I would imagine that if you are permitted to edit custom fields, you should be able to decide what fieldsets they belong in as well.
```
Yeah, you're probably right. It just feels kinda backwards to allow actions on fieldsets based on fields. Like editing a group because a can edit a user.

Reply:
I get that, but the fieldsets are really just containers, they don't specify permissions (as in your user/group analogy) . My concern here is always a UX/UI one. If people can easily grant one thing but forget to grant the other thing they realistically need in order to use the first thing, I try to avoid that. It ends up with frustrated users, since in a practical sense, no one would grant one without the other. Perhaps a simpler solution is to change the label in the permissions (not the value of the permission key) in the config file to "Custom fields and Fieldsets" to prevent any confusion.

Reply:
My issue is more with the naming in the code, because it seems unintuitive to read something like this:
```php
class CustomFieldsetsController extends Controller
{
    public function update($id)
    {
        $fieldset = CustomFieldset::find($id);
        $this->authorize('update', CustomField::class);
    }
}
```
while this seems fine
```php
class CustomFieldsController extends Controller
{
    public function destroy($id)
    {
        $field = CustomField::find($id);
        $this->authorize('update', CustomFieldset::class);
    }
}
```
because adding/updating/removing fields feels like updating the set.

Reply:
As I said, I understand your rationale, but ultimately, I believe in the BDD approach, solving the users problems first, even if that means the code needs a few extra comments to explain why we did a thing. 

I suppose a middle ground could be to put that logic in the `SnipePermissionsPolicy`, granting CustomFieldset access if Customfield access has been granted. Then the controllers make sense, and the core logic of granting both lives in the policy. I'd want a comment above that logic to remind us why we're doing that tho ;)